### PR TITLE
Randomize integration tests port assignments

### PR DESCRIPTION
**Pull request by xiangfu0**
Randomize pinot component ports to avoid port already bind issue

### CODE REVIEW

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/GrpcBrokerClusterIntegrationTest.java [hunk 1]
```diff
@@ -139,6 +139,7 @@ public void tearDown()
     stopServer();
     stopBroker();
     stopController();
+    stopKafka();
```
Comment (walterddr) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/GrpcBrokerClusterIntegrationTest.java:142]: LOL

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java [hunk 2]
```diff
@@ -392,7 +392,8 @@ public void addFakeServerInstancesToAutoJoinHelixCluster(int numInstances, boole
   public void addFakeServerInstancesToAutoJoinHelixCluster(int numInstances, boolean isSingleTenant, int baseAdminPort)
       throws Exception {
     for (int i = 0; i < numInstances; i++) {
-      addFakeServerInstanceToAutoJoinHelixCluster(SERVER_INSTANCE_ID_PREFIX + i, isSingleTenant, baseAdminPort + i);
+      addFakeServerInstanceToAutoJoinHelixCluster(SERVER_INSTANCE_ID_PREFIX + i, isSingleTenant,
+          NetUtils.findOpenPort(baseAdminPort + i));
```
Comment (walterddr) [inline pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java:396]: nit: if one port is unavailable netUtils.findOpenPort will increase by 1 --> which essentially makes the next instance i conflict and then tries to increase that as well. should we do truly randomize? what's limiting us to keep using the base + i format?

**Review by walterddr — COMMENTED** [review]
lgtm. few questions

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java (hunk 2 above)
Comment (xiangfu0) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java:396]: This method finds the open port then create the fake instance, so it won't happen.

Comment (xiangfu0) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java]: This method finds the open port then create the fake instance, so it won't happen.

Comment (xiangfu0) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java]: This method finds the open port then create the fake instance, so it won't happen.

Comment (walterddr) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java]: that's even worse right? if one admin port is available and one fake server took it (but not open the port); the next server will took the +1 port as well b/c it is still available --> that means 2 fake server will have the same admin port ?

Comment (xiangfu0) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java]: Make those ports purely randomized.

**Review by walterddr — APPROVED** [review]

**Pull request merged**